discovery_engine_core/core/src/engine.rs

```diff
@@ -321,15 +321,15 @@ impl Engine {
         // initialize the states
         #[cfg(feature = "storage")]
         let storage = {
-            let sqlite_uri = if config.use_in_memory_db {
-                "sqlite::memory:".into()
-            } else {
-                let db_path = PathBuf::from(&config.data_dir).join("db.sqlite");
-                format!("sqlite:{}", db_path.display())
-            };
-            let storage = SqliteStorage::connect(&sqlite_uri).await?;
-            storage.init_database().await?;
-            Box::new(storage)
+            let db_file_path = (!config.use_ephemeral_db).then(|| {
+                PathBuf::from(&config.data_dir).join("db.sqlite")
+                        .into_os_string()
+                        .into_string()
+                        .unwrap(/*can't fail as we only join rust strings*/)
+            });
+            let (storage, _init_hint) = SqliteStorage::init_storage_system(db_file_path).await?;
+            //FIXME pass the hint to dart for deciding if migrations are needed
```

Comment:
do we have a task to track this?

Reply:
yes the task from which this is the first (of multiple, likely 3) PRs :wink: 

Split because this is the first.